### PR TITLE
Adding json formatting to `--list-tags` option in `podman search` command.

### PR DESCRIPTION
**Pull request by Afourcat**
Data is formatted following this JSON structure:
```json
{
    "Name": "...",
    "Tags": ["...", "...", "..."]
}
```

Closes: #8740.

Signed-off-by: Alexandre Fourcat <afourcat@gmail.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @Afourcat 
LGTM
@mheon @Luap99 @vrothberg @containers/podman-maintainers PTAL

File: cmd/podman/images/search.go [hunk 1]
```diff
@@ -180,3 +179,39 @@ func imageSearch(cmd *cobra.Command, args []string) error {
...
+	}
+	fmt.Println(string(prettyJSON))
+	return nil
+}
+
+func buildListTagsJson(searchReport []entities.ImageSearchReport) interface{} {
+	entries := []struct {
+		Name string
+		Tags []string
+	}{}
```
Comment (Luap99) [inline cmd/podman/images/search.go:196]: Make a type definition for the struct outside of the function. Like this:
```golang
type listTagEntry struct {
    Name string
    Tags []string
}
```
Use this type for `entries`, `newElem` and also as return value for the function. `interface{}` should be avoided.

File: test/e2e/search_test.go [hunk 2]
```diff
@@ -124,6 +124,15 @@ registries = ['{{.Host}}:{{.Port}}']`
 		Expect(search.OutputToString()).To(ContainSubstring("docker.io/library/alpine"))
 	})
 
+	It("podman search format json list tags", func() {
+		search := podmanTest.Podman([]string{"search", "--list-tags", "--format", "json", "alpine"})
+		search.WaitWithDefaultTimeout()
+		Expect(search.ExitCode()).To(Equal(0))
+		Expect(search.IsJSONOutputValid()).To(BeTrue())
+		Expect(search.OutputToString()).To(ContainSubstring("docker.io/library/alpine"))
+		Expect(search.OutputToString()).To(ContainSubstring("3.10"))
```
Comment (Luap99) [inline test/e2e/search_test.go:133]: Maybe also test for a second tag.

**Review by Luap99 — COMMENTED** [review]
Thx @Afourcat
Two small nits besides that LGTM

File: cmd/podman/images/search.go (hunk 1 above)
Comment (Afourcat) [reply cmd/podman/images/search.go:196]: Ok, thank you for your advices.

File: test/e2e/search_test.go (hunk 2 above)
Comment (Afourcat) [reply test/e2e/search_test.go:133]: Ok, thank you.

**Comment by rhatdan** [comment]
@Afourcat please squash and sign your commits.

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks for contributing!

**Comment by vrothberg** [comment]
I started the flaked test

**Comment by edsantiago** [comment]
/hold cancel 

Nice work, thanks again @Afourcat .

**Comment by Afourcat** [comment]
Thank you for your work.

**Pull request merged**